### PR TITLE
feat(ncyBreadcrumbText): Add ncyBreadcrumbText directive

### LINKED ISSUE #71: Title tag with full state chain
It would be nice to have a directive which we can use with a title tag, to put the full state chain into the page title.  Something like ncy-breadcrumb-last, except it shows the entire state chain separated by a separator string.

Right now, I'm getting by with a custom bit of code:

````
    <title ng-bind="'MyApp: ' + titleTag">MyApp</title>
````

````
        $rootScope.$on('$viewContentLoaded', function () {
            BuildTitleTag();
        });

        function BuildTitleTag()
        {
            var viewScope = $breadcrumb.$getLastViewScope();
            var steps = $breadcrumb.getStatesChain();
            var combinedLabel = '';
            angular.forEach(steps, function (step) {
                if (step.ncyBreadcrumb && step.ncyBreadcrumb.label) {
                    var parseLabel = $interpolate(step.ncyBreadcrumb.label);
                    combinedLabel += " / " + parseLabel(viewScope);
                } else {
                    combinedLabel += " / " + step.name;
                }
            });

            if (combinedLabel.length > 3) {
                combinedLabel = combinedLabel.substring(3);
            }

            $rootScope.titleTag = combinedLabel;
        }

````

### PR DESCRIPTION
**Pull request by TravisTX**
Closes #71

### CODE REVIEW

File: dist/angular-breadcrumb.js [hunk 1]
```diff
@@ -284,8 +284,95 @@ function BreadcrumbLastDirective($interpolate, $breadcrumb, $rootScope) {
...
+            return {
+                post: function postLink(scope) {
+                    var labelWatchers = [];
+                    
+                    var registerWatchersText = function(labelWatcherArray, interpolationFunction, viewScope, step) {
+                        angular.forEach(getExpression(interpolationFunction), function(expression) {
+                            var watcher = viewScope.$watch(expression, function() {
+                                step.ncyBreadcrumbLabel = interpolationFunction(viewScope);
+                                renderLabel();
+                            });
```
Comment (ncuillery) [inline dist/angular-breadcrumb.js]: We build the entire chain of states in the `renderLabel`, so specific step data is not used. L313 is not needed and the whole watch could be reduced to:
```javascript
var watcher = viewScope.$watch(expression, renderLabel);
```

File: dist/angular-breadcrumb.js [hunk 2]
```diff
@@ -284,8 +284,95 @@ function BreadcrumbLastDirective($interpolate, $breadcrumb, $rootScope) {
...
+                                combinedLabel += separator + step.name;
+                            }
+                        });
+
+                        // strip leading separator
+                        if (combinedLabel.length > separator.length) {
+                            combinedLabel = combinedLabel.substring(separator.length);
+                        }
+
+                        scope.ncyBreadcrumbLabel = combinedLabel;
```
Comment (ncuillery) [inline dist/angular-breadcrumb.js]: We can take advantage of native Array methods here:

```javascript
var combinedLabels = [];
angular.forEach(steps, function (step) {
  if (step.ncyBreadcrumb && step.ncyBreadcrumb.label) {
    var parseLabel = $interpolate(step.ncyBreadcrumb.label);
    combinedLabels.push(parseLabel(viewScope));
  } else {
    combinedLabels.push(step.name);
  }
});

scope.ncyBreadcrumbLabel = combinedLabels.join(separator);
```

File: dist/angular-breadcrumb.js [hunk 3]
```diff
@@ -284,8 +284,95 @@ function BreadcrumbLastDirective($interpolate, $breadcrumb, $rootScope) {
...
+                    };
+
+                    $rootScope.$on('$viewContentLoaded', function () {
+                        setupWatchers();
+                        renderLabel();
+                    });
+
+                    // View(s) may be already loaded while the directive's linking
+                    setupWatchers();
+                    renderLabel();
```
Comment (ncuillery) [inline dist/angular-breadcrumb.js]: Multiple performance-consuming instructions are duplicated in `setupWatchers` and `renderLabel`:
- Extract chain of states from the service
- Loop on the states
- Interpolation of labels

Couldn't we merge `setupWatchers` in `renderLabel`  ? (look at the ncyBreadcrumbLast directive)

File: dist/angular-breadcrumb.js (hunk 2 above)
Comment (ncuillery) [inline dist/angular-breadcrumb.js]:
```javascript
scope.ncyBreadcrumbChain = combinedLabels.join(separator);
```
of course :wink:

**Comment by ncuillery** [comment]
Thanks a lot for the PR :-) 

I just noticed some problems whose can be fix easily thanks to your solid test suite. Please, can you fix it ?

**Comment by TravisTX** [comment]
Workin on it now.  Thanks for the review.

**Comment by ncuillery** [comment]
Landed, thanks a lot !

**Pull request merged**